src/azure-cli/azure/cli/command_modules/synapse/operations/sqlpoolblobauditingpolicy.py

```diff
@@ -0,0 +1,229 @@
+        storage_account=None,
+        storage_endpoint=None,
+        storage_account_access_key=None,
+        retention_days=None,
+        audit_actions_and_groups=None,
+        storage_account_subscription_id=None,
+        is_storage_secondary_key_in_use=None,
+        is_azure_monitor_target_enabled=None):
+    '''
+    Updates a sql pool blob auditing policy. Custom update function to apply parameters to instance.
+    '''
+
+    # Validate input arguments
+    blob_storage_arguments_provided = (
+        storage_account is not None or
+        storage_endpoint is not None or
+        storage_account_access_key is not None or
+        retention_days is not None)
+
+    if not state and not blob_storage_arguments_provided:
+        raise CLIError('Either state or blob storage arguments are missing')
+
+    if retention_days is not None and (not retention_days.isdigit() or int(retention_days) <= 0):
+        raise CLIError('retention-days must be a positive number greater than zero')
+
+    if state is not None:
+        instance.state = BlobAuditingPolicyState[state.lower()]
+
+    if _is_audit_policy_state_enabled(instance.state):
+        if is_storage_secondary_key_in_use is not None:
+            instance.is_storage_secondary_key_in_use = is_storage_secondary_key_in_use
+
+        if is_azure_monitor_target_enabled is not None:
+            instance.is_azure_monitor_target_enabled = is_azure_monitor_target_enabled
+
+        # handle storage account and storage endpoint
+        _audit_policy_update_apply_blob_storage_details(cmd, instance, retention_days, storage_account,
+                                                        storage_account_access_key, storage_endpoint,
+                                                        storage_account_subscription_id)
+
+        if audit_actions_and_groups is not None:
+            instance.audit_actions_and_groups = audit_actions_and_groups
+
+        if not instance.audit_actions_and_groups or instance.audit_actions_and_groups == []:
+            instance.audit_actions_and_groups = [
+                "SUCCESSFUL_DATABASE_AUTHENTICATION_GROUP",
+                "FAILED_DATABASE_AUTHENTICATION_GROUP",
+                "BATCH_COMPLETED_GROUP"]
+
+    return instance
+
+
+def _audit_policy_update_apply_blob_storage_details(cmd, instance, retention_days, storage_account,
+                                                    storage_account_access_key, storage_endpoint,
+                                                    storage_account_subscription_id):
+    if storage_account is not None:
+        storage_resource_group = _find_storage_account_resource_group(cmd.cli_ctx, storage_account)
+        storage_endpoint = _get_storage_endpoint(cmd.cli_ctx, storage_account, storage_resource_group)
+        storage_account_subscription_id = _find_storage_account_subscription_id(cmd.cli_ctx, storage_account)
+
+    if storage_endpoint is not None:
+        instance.storage_endpoint = storage_endpoint
```

Comment:
Also need to add this?
```   
# Validate storage endpoint arguments
if storage_endpoint and storage_account:
    raise CLIError('--storage-endpoint and --storage-account cannot both be specified.')
```

Reply:
Sql db doesn't add this validation. I want to keep the same with sql db although it will not cause error without this validation.

---
In reply to: [533155346](https://github.com/Azure/azure-cli/pull/16095#discussion_r533155346) [](ancestors = 533155346)